### PR TITLE
Add stylelint

### PR DESCRIPTION
**Pull request by jum-s**
After quite some time trying to configure [prettier](https://prettier.io), i can say that their non-modifiable opinionated configuration is bad (im not the only one [complaining](https://github.com/prettier/prettier-vscode/issues/352) ).

So instead, the last commits setup and fix with stylelint. Rules are soft, as CSS readability is less important than JS).

### CODE REVIEW

**Comment by maxlath** [comment]
@jum-s it seems that the commit that added `scripts/stylelint.cjs` disappeared

**Comment by jum-s** [comment]
I dont think there was any integrated scripts before. IDE extension use default config file.
I added something in 338e4f39f but not sure its worth the time spent.

**Comment by maxlath** [comment]
I think it's useful to have it in pre-commit hooks, similarly to eslint, or generally to have it as an available script to be easily run on the whole codebase, in case we change some rules in the future.

File: scripts/lint_fix [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env sh
 if [ "$1" != "" ] ; then
   eslint --config .eslintrc.cjs --format codeframe --ext js,svelte --fix "$@"
+  stylelint --config .stylelintrc.cjs --fix "$@/**/*.svelte"
 else
   eslint --config .eslintrc.cjs --format codeframe --ext js,svelte --fix app tests scripts
+  stylelint --config .stylelintrc.cjs --fix "**/*.svelte"
```
Comment (maxlath) [inline scripts/lint_fix:7]: same as above
```suggestion
  stylelint --config .stylelintrc.cjs --fix app
```

File: scripts/lint [hunk 2]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env sh
 if [ "$1" != "" ] ; then
   eslint --config .eslintrc.cjs --format codeframe --ext js,svelte "$@"
+  stylelint --config .stylelintrc.cjs "$@/**/*.svelte"
 else
   eslint --config .eslintrc.cjs --format codeframe --ext js,svelte app tests scripts
+  stylelint --config .stylelintrc.cjs "**/*.svelte"
```
Comment (maxlath) [inline scripts/lint:7]: and again, let the `.stylelintignore` determine what should be checked
```suggestion
  stylelint --config .stylelintrc.cjs app
```

File: package.json [hunk 3]
```diff
@@ -114,13 +114,18 @@
     "mocha": "^9.2.0",
     "node-fetch": "^2.6.1",
     "postcss": "^8.2.4",
+    "postcss-html": "^1.5.0",
     "postcss-loader": "^4.2.0",
     "postcss-preset-env": "^6.7.0",
+    "postcss-scss": "^4.0.6",
```
Comment (maxlath) [inline package.json:120]: Is that package required? I tried removing it, and stylelint didn't complain.

Comment (jum-s) [reply package.json:120]: yes! Looks like its a double failing that works though : `post-css-scss` seems necessary to my config (testing by removing  `post-css-scss` and `stylelint-config-recommended-scss`), but `stylelint-config-recommended-scss` is bundling `post-css-scss`. That's why it's still working if you remove this line. I would be removing this line and for keeping  `stylelint-config-recommended-scss` as configuration does look fit to the codebase and bundles `postcss-scss`.

File: .stylelintrc.cjs [hunk 4]
```diff
@@ -0,0 +1,30 @@
+// This config file is used by stylelint
+// See package.json scripts: lint*
+// Rules documentation: https://stylelint.io/user-guide/rules/
+module.exports = {
+  extends: [
+    'stylelint-config-recommended-scss',
```
[the remark this answers is not in the document]
Comment (jum-s) [reply .stylelintrc.cjs:6]: see comment below about `postcss-scss`. I would be for keeping `stylelint-config-standard-scss`.

File: scripts/lint [hunk 5]
```diff
@@ -1,6 +1,8 @@
 #!/usr/bin/env sh
 if [ "$1" != "" ] ; then
   eslint --config .eslintrc.cjs --format codeframe --ext js,svelte "$@"
+  stylelint --config .stylelintrc.cjs "$@/**/*.svelte"
```
[the remark this answers is not in the document]
Comment (jum-s) [reply scripts/lint:4]: done in commit named "add stylelint to lint scripts"

File: .stylelintrc.cjs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    'stylelint-config-css-modules'
```
[the remark this answers is not in the document]
Comment (jum-s) [reply .stylelintrc.cjs:7]: > I guess this is to add support for :global pseudo-class?

yes.

I dont get why not using css modules, but i dont want to spend more time on this. Done in b7c66b662

File: .stylelintrc.cjs [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+  ],
+  rules: {
+    'at-rule-no-unknown': null,
+    'color-named': null,
+    'declaration-colon-space-after': 'always',
+    'block-opening-brace-newline-after': 'always-multi-line',
+    'function-no-unknown': null,
```
[the remark this answers is not in the document]
Comment (jum-s) [reply .stylelintrc.cjs:14]: done in 7567b12db

File: .stylelintrc.cjs [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+    'function-url-no-scheme-relative': true,
+    'function-url-quotes': 'always',
+    indentation: 2,
+    'no-descending-specificity': null,
+    'no-duplicate-selectors': true,
+    'scss/no-global-function-names': null,
+    'string-quotes': 'single',
```
[the remark this answers is not in the document]
Comment (jum-s) [reply .stylelintrc.cjs:21]: I removed this rule, as your suggestion updates dozens of documents for not much difference. Leaving it for a stricter PR later.

File: package.json (hunk 3 above)
Comment (jum-s) [reply package.json:120]: line removed in 3591f6ef1

**Comment by jum-s** [comment]
Hum, I was not expecting that in depth look at stylelint. Actually, im having second thought on this PR. The customization your mentioning in the comments above is quite some work for relatively small DX advantages. And its adding new conventions where, CSS looked readable to me beforehand.

I though i woulc introduce CSS linting smoothly/lightly, without enforcing much. That's why there was initially no scripts and there is still no `.scss` coverage. 

I pushed a few updates taking most of your comments above into account. But i would propose to leave it configured by default with `stylelint-config-standard-scss` setup. And maybe a more strict setup can wait a bit (or another PR)?

I wont do much more on this PR, and Im also fine not merging it, which would  leave stylelint as custom local settings, and it would be fine for me.

**Comment by maxlath** [comment]
@jum-s I do see quite some DX improvement from it. I can take over this PR if you lost interest, or merge it like this and propose further customization in a follow-up PR.

**Comment by jum-s** [comment]
Should be ready for review and merge. i might take a little while to review stylelint PR (as i had my share of it for now), but you are welcome to open another PR for some fine tuning.

**Pull request merged**